### PR TITLE
Inf table

### PR DESCRIPTION
**Pull request by dlrice**
Incorporate franklin-sites' new data-table which uses infinite loading.

* The url for the next page of results is extracted from the header.
* "Loading..." appears on new set of results (eg when a column is sorted).
* Increase zIndex of bug report

### CODE REVIEW

File: src/data/columnAttributes.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const small = 200; //10rem;
```
Comment (xwatkins) [inline src/data/columnAttributes.tsx]: This should be merged with `FieldToViewMappings.tsx` and renamed to something like `ResultsColumns` or `ColumnConfiguration`?

Comment (dlrice) [reply src/data/columnAttributes.tsx]: Nice idea. Done.

File: src/results/utils/utils.ts [hunk 2]
```diff
@@ -6,12 +6,13 @@ import {
   SelectedFacet,
 } from '../types/resultsTypes';
 
-export const createFacetsQueryString = (facets: SelectedFacet[]) =>
-  facets.reduce(
+export const createFacetsQueryString = (facets: SelectedFacet[]) => {
+  return facets.reduce(
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/results/utils/utils.ts]: Done.

File: src/results/state/actions.ts [hunk 3]
```diff
@@ -1,45 +1,104 @@
...
+    nextUrl,
+    totalNumberResults,
+    isNextPage,
   });
 
-export const fetchResultsRequest = () => action(FETCH_RESULTS_REQUEST);
+export const requestResults = (url: string) => action(REQUEST_RESULTS, { url });
 
-export const fetchResults = (
+/*
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/results/state/actions.ts]: Nope.

File: src/results/state/actions.ts [hunk 4]
```diff
@@ -1,45 +1,104 @@
...
-  fetchData(
-    getAPIQueryUrl(queryString, columns, selectedFacets, sortBy, sortDirection)
-  ).then(response => dispatch(fetchResultsSuccess(response.data)));
+*/
+
+const getNextUrlFromResponse = (link: string | null) => {
+  if (!link) {
+    return;
+  }
+  const re = /<([0-9a-zA-Z$\-_.+!*'(),?\/:=&%]+)>; rel="next"/;
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/results/state/actions.ts]: Creating a jira for this.

File: src/results/state/actions.ts [hunk 5]
```diff
@@ -1,45 +1,104 @@
...
+  const match = re.exec(link);
+  return match && match[1];
+};
+
+export const clearResults = () => action(CLEAR_RESULTS);
+
+export const fetchResults = (url: string, isNextPage: boolean) => async (
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/results/state/actions.ts]: Done.

File: src/results/state/actions.ts [hunk 6]
```diff
@@ -1,45 +1,104 @@
...
+  dispatch: Dispatch
+) => {
+  if (!isNextPage) {
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/results/state/actions.ts]: Done.

File: src/results/state/actions.ts [hunk 7]
```diff
@@ -1,45 +1,104 @@
...
   // .catch(error => console.error(error)); // the console creates a tslint ...
   // ... error but we want to catch this in the future
 };
 
+export const shouldFetchResults = (url: string, state: RootState) => {
+  const { isFetching, isFetched } = state.results;
+  return !isFetching[url] && !isFetched[url];
+};
+
+export const fetchResultsIfNeeded = (
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/results/state/actions.ts]: Done.

File: src/results/state/reducers.ts [hunk 8]
```diff
@@ -1,9 +1,25 @@
 import { array } from 'prop-types';
 import { ActionType } from 'typesafe-actions';
+import queryString from 'query-string';
 import * as resultsActions from './actions';
 import initialState, { ResultsState } from './initialState';
 import { SortDirections, SortDirectionsType } from '../types/resultsTypes';
 
+// const isSameSearch = (urls, newUrl) => {
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/results/state/reducers.ts]: Nope. Removed.

File: src/results/state/reducers.ts [hunk 9]
```diff
@@ -20,19 +36,39 @@ const results = (state: ResultsState = initialState, action: ResultAction) => {
...
       return {
         ...state,
-        results: action.payload.data.results,
+        results: [...state.results, ...action.payload.data.results],
         facets: action.payload.data.facets,
         lastUpdated: action.payload.receivedAt,
-        isFetching: false,
+        totalNumberResults: action.payload.totalNumberResults,
+        isFetching: {
+          ...state.isFetching,
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/results/state/reducers.ts]: Yes. Done.

**Pull request merged**